src/entities/swap/index.ts

```diff
@@ -2,36 +2,31 @@ import { TokenAmount } from '../tokenAmount';
 import { SwapKind } from '../../types';
 import { PriceImpactAmount } from '../priceImpactAmount';
 import {
-    Path,
     SwapBase,
     SwapCallExactIn,
     SwapCallExactOut,
     SwapBuildOutputExactIn,
     SwapBuildOutputExactOut,
+    SwapInput,
 } from './types';
 import { SwapV2 } from './swapV2';
 import { validatePaths } from './pathHelpers';
 import { Slippage } from '../slippage';
+import { SwapV3 } from './swapV3';
 
 export * from './types';
 
 // A Swap can be a single or multiple paths
 export class Swap {
     private readonly swap: SwapBase;
 
-    public constructor({
-        chainId,
-        paths,
-        swapKind,
-    }: { chainId: number; paths: Path[]; swapKind: SwapKind }) {
-        validatePaths(paths);
+    public constructor(swapInput: SwapInput) {
+        validatePaths(swapInput.paths);
 
-        switch (paths[0].balancerVersion) {
-            case 2:
-                this.swap = new SwapV2({ chainId, paths, swapKind });
-                return;
-            case 3:
-                throw new Error('Unsupported Balancer Protocol Version');
+        if ('wethIsEth' in swapInput) {
```

Comment:
On Add/Remove Liquidity implementation we have the same external interface for both v2 and v3, where we expose a flag that the user specifies if the input is ETH or not.
Currently `useNativeAssetAsWrappedAmountIn`, but soon to be refactored to be consistent with `wethIsEth` from SC.
Internally we check that flag and change addresses to ADDRESS_ZERO accordingly.
If we follow the same pattern, we should ideally expose the same flag for swapV2 and update addresses internally.
Checking for `wethIsEth` flag wouldn't work on this scenario.
A suggestion would be to use `balancerVersion` available within `swapInput.paths[0]` instead.

Reply:
Ok, yeah I think this sounds good. In this case we will expect that the consumer always uses WETH for Paths and has no concept of using ZERO address?

Reply:
Yeap. Zero address is an implicit requirement that we want to avoid. Having an explicit flag should be a better experience.

Reply:
I refactored this to move wethIsEth to buildCall. As discussed with Daniel, when a user wants to swap using native asset they will query the API using the wrapped token. They can then decide to user wrapped/native by setting the param in buildCall.